android/src/main/java/org/reactnative/camera/tasks/ResolveTakenPictureAsyncTask.java

```diff
@@ -44,99 +44,109 @@ public ResolveTakenPictureAsyncTask(byte[] imageData, Promise promise, ReadableM
+                    mBitmap = rotateBitmap(mBitmap, getImageRotation(orientation));
+                }
+
+                if (mOptions.hasKey("mirrorImage") && mOptions.getBoolean("mirrorImage")) {
+                    mBitmap = flipHorizontally(mBitmap);
+                }
+
+                // Write Exif data to the response if requested
+                if (mOptions.hasKey("exif") && mOptions.getBoolean("exif")) {
+                    WritableMap exifData = RNCameraViewHelper.getExifData(exifInterface);
+                    response.putMap("exif", exifData);
+                }
+            }
+
+            // Upon rotating, write the image's dimensions to the response
+            response.putInt("width", mBitmap.getWidth());
+            response.putInt("height", mBitmap.getHeight());
+
+            // Cache compressed image in imageStream
+            ByteArrayOutputStream imageStream = new ByteArrayOutputStream();
+            mBitmap.compress(Bitmap.CompressFormat.JPEG, getQuality(), imageStream);
+
+            // Write compressed image to file in cache directory
+            String filePath = writeStreamToFile(imageStream);
+            File imageFile = new File(filePath);
+            String fileUri = Uri.fromFile(imageFile).toString();
+            response.putString("uri", fileUri);
+
+            // Write base64-encoded image to the response if requested
+            if (mOptions.hasKey("base64") && mOptions.getBoolean("base64")) {
+                response.putString("base64", Base64.encodeToString(imageStream.toByteArray(), Base64.DEFAULT));
+            }
+
+            // Cleanup
+            imageStream.close();
+            if (inputStream != null) {
+                inputStream.close();
+                inputStream = null;
+            }
+
+            return response;
+        } catch (Resources.NotFoundException e) {
+            mPromise.reject(ERROR_TAG, "Documents directory of the app could not be found.", e);
+            e.printStackTrace();
+        } catch (IOException e) {
+            mPromise.reject(ERROR_TAG, "An unknown I/O exception has occurred.", e);
+            e.printStackTrace();
+        } finally {
+            try {
+                if (inputStream != null) {
+                    inputStream.close();
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        // An exception had to occur, promise has already been rejected. Do not try to resolve it again.
+        return null;
+    }
 
     private Bitmap rotateBitmap(Bitmap source, int angle) {
         Matrix matrix = new Matrix();
         matrix.postRotate(angle);
         return Bitmap.createBitmap(source, 0, 0, source.getWidth(), source.getHeight(), matrix, true);
     }
 
+    private Bitmap resizeBitmap(Bitmap bm, int newWidth) {
```

Comment:
Last change here, adding the `resize`method. Any other change in the file is related to the indent failure